chessboard/chessboard.go

```diff
@@ -0,0 +1,48 @@
+package chessboard
+
+type Rank []bool
+
+type Chessboard map[string]Rank
+
+// CountInRank returns how many squares are occupied in the chessboard,
+// within the given rank
+func CountInRank(cb Chessboard, rank string) (count int) {
+	for i := range cb[rank]{
+		if cb[rank][i]{count += 1}
+	}
+	return
+}
+
+// CountInFile returns how many squares are occupied in the chessboard,
+// within the given file
+func CountInFile(cb Chessboard, file int) (count int) {
+	if file < 0 || file>len(cb){
+		return 0
+	}
+	for i := range cb {
+		if cb[i][file-1]{count += 1}
+	}
+	return
+}
+
+// CountAll should count how many squares are present in the chessboard
+func CountAll(cb Chessboard) (count int) {
+	for _,r := range cb{
```

Comment:
You can declare Rank with a size and replace all of this with `return len(cb) * 8`

Reply:
You're correct. What if you were playing some variation of chess that used some other square (or even rectangular) square count?

Reply:
I wouldn't call it "Chessboard" :stuck_out_tongue:

Reply:
I mean, [there seems to be some variations on what constitutes a chessboard](https://greenchess.net/variants.php?cat=2) ;)

Reply:
I think this is the worst assignment so far. Not your implementation, @robnester-rh , but the assignment itself.

Reply:
The f**k. I had no idea about those variations. Nice.